src/shell/interpreter.zig

```diff
@@ -936,100 +1158,87 @@ pub const Interpreter = struct {
-        var jsstrings = std.ArrayList(bun.String).initCapacity(stack_alloc.get(), 4) catch {
-            globalThis.throwOutOfMemory();
-            return null;
+        const resolve = arguments.nextEat() orelse {
+            globalThis.throw("shell: expected 3 arguments, got 0", .{});
+            return .undefined;
         };
-        defer {
-            for (jsstrings.items[0..]) |bunstr| {
-                bunstr.deref();
-            }
-            jsstrings.deinit();
-        }
-        var jsobjs = std.ArrayList(JSValue).init(arena.allocator());
-        var script = std.ArrayList(u8).init(arena.allocator());
-        if (!(bun.shell.shellCmdFromJS(globalThis, string_args, template_args, &jsobjs, &jsstrings, &script) catch {
-            globalThis.throwOutOfMemory();
-            return null;
-        })) {
-            return null;
-        }
-
-        var parser: ?bun.shell.Parser = null;
-        var lex_result: ?shell.LexResult = null;
-        const script_ast = ThisInterpreter.parse(
-            &arena,
-            script.items[0..],
-            jsobjs.items[0..],
-            jsstrings.items[0..],
-            &parser,
-            &lex_result,
-        ) catch |err| {
-            if (err == shell.ParseError.Lex) {
-                assert(lex_result != null);
-                const str = lex_result.?.combineErrors(arena.allocator());
-                globalThis.throwPretty("{s}", .{str});
-                return null;
-            }
 
-            if (parser) |*p| {
-                if (bun.Environment.allow_assert) {
-                    assert(p.errors.items.len > 0);
-                }
-                const errstr = p.combineErrors();
-                globalThis.throwPretty("{s}", .{errstr});
-                return null;
-            }
+        const reject = arguments.nextEat() orelse {
+            globalThis.throw("shell: expected 3 arguments, got 0", .{});
+            return .undefined;
+        };
 
-            globalThis.throwError(err, "failed to lex/parse shell");
-            return null;
+        const parsed_shell_script_js = arguments.nextEat() orelse {
+            globalThis.throw("shell: expected 3 arguments, got 0", .{});
+            return .undefined;
         };
 
-        const script_heap = arena.allocator().create(bun.shell.AST.Script) catch {
-            globalThis.throwOutOfMemory();
-            return null;
+        const parsed_shell_script = parsed_shell_script_js.as(ParsedShellScript) orelse {
+            globalThis.throw("shell: expected a ParsedShellScript", .{});
+            return .undefined;
         };
 
-        script_heap.* = script_ast;
+        var shargs: *ShellArgs = undefined;
+        var jsobjs: std.ArrayList(JSValue) = std.ArrayList(JSValue).init(allocator);
+        var quiet: bool = false;
+        var cwd: ?bun.String = null;
+        var export_env: ?EnvMap = null;
+
+        parsed_shell_script.take(
```

Comment:
What verifies that the shell args in `parsed_shell_scripts` is not null? If it was null, wouldn't this be undefined memory and cause a segfault?